v1/topdown/cache.go

```diff
@@ -32,6 +32,12 @@ type VirtualCache interface {
 	Keys() []ast.Ref
 }
 
+// BaseCache defines the interface for a cache that stores cached base documents, i.e. data.
+type BaseCache interface {
+	Get(ref ast.Ref) ast.Value
+	Put(ref ast.Ref, value ast.Value)
```

Comment:
[nit] since it's optional and doesn't really add anything, let's drop the names
```suggestion
	Get(ast.Ref) ast.Value
	Put(ast.Ref, ast.Value)
```

Reply:
Fair enough.